examples/progress_bar_minimal.py

```diff
@@ -34,7 +34,12 @@ def iterable_w_context():
             # using a context manager also allows us to manipulate
             # the progress object e.g. by setting a description
             pbr.set_description(f"Slice {i}")
-            process(im_slice)
+
+            # a context manager is also the only way to make use
+            # of nested progress bars and have them aligned in
+            # activity dock nicely
+            for channel in progress(im_slice):
+                process(channel)
```

Comment:
actually, If I click the "iterable with context" button twice, I also get this error when the second one finishes.  I think there might be too many things trying to close it (with the `Qt.WA_DeleteOnClose` flag set as well)

Reply:
Hmmm so our context var approach isn't really detecting nesting specifically, and that's what's leading to these errors I think. 

It's detecting if there's already a context present. So, looking at the minimal examples again, we can click "Iterable" all day long, and that will be fine.

But, if we click "Iterable with Context" twice, our `CURRENT_GROUP` will be set already, and we'll see this as a nested context, even though it's actually just two separate contexts.
This is also true with "Indeterminate" because we're using it in a context.

This is particularly noticeable now that we don't always have a `ProgressBarGroup`. With "Indeterminate", clicking it a second time results in a "ProgressBarGroup" being created. When one of these bars hits 42, it closes the whole group, which also closes the progress bar that was still going.

It feels like with this approach we'd need some sort of GUID (or at least UID) for the context var, so we can detect not only if we've entered the context, but also if we're creating a new progress bar inside the same context. Does this sound sane at all?

Reply:
> It feels like with this approach we'd need some sort of GUID (or at least UID) for the context var, so we can detect not only if we've entered the context, but also if we're creating a new progress bar inside the same context

Of course, the issue with this approach is we only enter the context after the `ProgressBar`'s been built, so we'd need to move the nesting logic into the `__enter__` method which is potentially quite messy.

Reply:
Upon further consideration, I don't know if we can achieve what we're trying to achieve here.

The critical issue is that our code is not proceeding procedurally, but is rather event driven. This means that at any point (even while a context is currently being executed), I can click a button and trigger another progress bar to be created.

The creation of this new progress bar could trigger the creation of a new context, and we have no way of knowing whether this new context is an instance of nesting, or just a new context being created somewhere randomly in some code (potentially not even in the same file as our original context).

This means we can't rely on class variables or context variables to tell us whether we're nested, the way we usually might. 

For example, one option I considered was incrementing a count context variable when we first enter a context, and then in future `__enter__`s, if this is >0 then we're nesting, because another context must've been created before us. We can then decrement count in `__exit__`. But this won't work because if I click two buttons in `progress_minimal.py`, and both use contexts, e.g. clicking "Iterable With Context" and "Indeterminate", then checking the count of the context variable will show me 1 - the context of whichever button I clicked first.

I also thought of using `tqdm._instances` - the `WeakSet` used by `tqdm` to track current active objects, but this would fail for the same reason - we can't assume having two active instances of `progress` is due to nesting.

So I'm a little bit stuck here. @tlambert03 any ideas?